plugins/security/nexus-ldap-client/src/main/java/org/sonatype/security/realms/ldap/client/Connection.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.security.realms.ldap.client;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * LDAP Server connection configuration.
+ *
+ * @sinnce 3.0
+ */
+public final class Connection
```

Comment:
Current NX uses two rest implementations: legacy Restlet (to be removed) and Siesta based (aka "modern").

All the `-client` modules (those that are also Java6) that use legacy Restlet endpoints, also define along service ifaces also VOs, to not have client consumer rely on legacy DTOs. Basically, client "hides" the actual REST DTOs from integrators, on purpose: Restlet API is legacy, way old and about to be removed. Look at all the client modules, none of them "leak" Restlet/legacy DTOs.

True, in this case yes, I did copy+paste the internal packages, but they are not the "same" (as data travels as nx-internal > legacy-restlet > legacy-dtos-over-wire > client) and client "repacks" them from restlet DTOs. Later, internal packages might change, but if the REST service (legacy restlet, or whatever will existing then) does not reflect that internal change, there is no need to "track" changes on client side either.

While this does look as violation of DRY, the purpose of it -- as is in all -client modules -- is to protect clients from leak of deprecated, to be removed DTOs. Similarly, sending internal packages down the wire would prevent internal changes without affecting clients.